public/app/features/alerting/unified/components/rules/central-state-history/EventDetails.tsx

```diff
@@ -0,0 +1,209 @@
+    const ruleUID = record.line?.ruleUID ?? '';
+    const identifier = useMemo(() => {
+        return parse(ruleUID, true);
+    }, [ruleUID]);
+    const { error, loading, result: rule } = useCombinedRule({ ruleIdentifier: identifier });
+
+    if (error) {
+        return (
+            <Text>
+                <Trans i18nKey="central-alert-history.details.error">Error loading rule</Trans>
+            </Text>
+        );
+    }
+    if (loading) {
+        return (
+            <Text>
+                <Trans i18nKey="central-alert-history.details.loading">Loading...</Trans>
+            </Text>
+        );
+    }
+
+    if (!rule) {
+        // if we get here assume we can't find the rule
+        return (
+            <Text>
+                <Trans i18nKey="central-alert-history.details.not-found">Rule not found</Trans>
+            </Text>
+        );
+    }
+    const getTransitionsCountByRuleUID = (ruleUID: string) => {
+        return logRecords.filter((record) => record.line.ruleUID === ruleUID).length;
+    };
+    return (
+        <Stack direction="column" gap={0.5}>
+            <Stack direction={'row'} gap={6}>
+                <StateTransition record={record} />
+                <ValueInTransition record={record} />
+                <NumberTransitions transitions={ruleUID ? getTransitionsCountByRuleUID(ruleUID) : 0} />
+            </Stack>
+            <Annotations rule={rule} />
+            <QueryVizualization rule={rule} ruleUID={ruleUID} logRecords={logRecords} />
+        </Stack>
+    );
+}
+
+function StateTransition({ record }: { record: LogRecord }) {
+    return (
+        <Stack gap={0.5} direction={'column'}>
+            <Text variant="body" weight="light" color="secondary">
+                <Trans i18nKey="central-alert-history.details.state-transitions">State transition</Trans>
+            </Text>
+            <Stack gap={0.5} direction={'row'} alignItems="center">
+                <EventState state={record.line.previous} showLabel />
+                <Icon name="arrow-right" size="lg" />
+                <EventState state={record.line.current} showLabel />
+            </Stack>
+        </Stack>
+    );
+}
+
+const Annotations = ({ rule }: { rule: CombinedRule }) => {
+    const styles = useStyles2(getStyles);
+    const annotations = rule.annotations;
+    if (!annotations) {
+        return null;
+    }
+
+    return (
+        <>
+            <Text variant="body" color="secondary" weight="light">
+                Annotations
+            </Text>
+            {Object.keys(annotations).length === 0 ? (
+                <Text variant="body" weight="light" italic>
+                    No annotations
+                </Text>
+            ) : (
+                <div className={styles.metadataWrapper}>
+                    {Object.entries(annotations).map(([name, value]) => (
+                        <MetaText direction="column" key={name}>
```

Comment:
same here, `Metatext` component render exaclty what I need, so I decided to reuse it